src/main/scala/chisel3/tester/ChiselTester.scala

```diff
@@ -0,0 +1,73 @@
+// See LICENSE for license details.
+
+package chisel3.tester
+
+import chisel3.experimental.MultiIOModule
+import firrtl.ExecutionOptionsManager
+import org.scalatest._
+import org.scalatest.exceptions.TestFailedException
+
+import scala.collection.mutable
+import scala.util.DynamicVariable
+
+/**
+  * Used to run simple tests that do not require a scalatest environment in order to run
+  * @param testName This will be used to generate a working directory in ./test_run_dir
+  */
+private class ChiselTester(testName: String) extends Assertions with TestEnvInterface {
+  // Provide test fixture data as part of 'global' context during test runs
+  val topFileName = Some(testName)
+
+  private def runTest[T <: MultiIOModule](tester: BackendInstance[T])(testFn: T => Unit) {
+    batchedFailures.clear()
+
+    Context.run(tester, this, testFn)
+  }
+
+  def getTestOptions: TesterOptions = {
+    TesterOptions(topFileName.get, writeVcd = false)
+  }
+
+  // This should be the only user-called function
+  def test[T <: MultiIOModule](dutGen: => T)(testFn: T => Unit) {
+    runTest(Context.createDefaultTester(() => dutGen, getTestOptions, None))(testFn)
+  }
+
+  def test[T <: MultiIOModule](dutGen: => T, execOptions: ExecutionOptionsManager)(testFn: T => Unit) {
+    runTest(Context.createDefaultTester(() => dutGen, getTestOptions, Some(execOptions)))(testFn)
+  }
+}
+
+/**
+  * This is a simple tester that does not require that it be within the scope of a scalatest
+  * in order to run. This form is suitable for running in the Jupyter notebook.
+  *
+  * @todo When Phases, Stages is implemented add ability to change testing options.
+  */
+object ChiselTester {
+
+  private var count = 0
```

Comment:
What's this for?

Reply:
Intended to strenghten guarantee of unique name.

Reply:
I'd prefer this was eliminated entirely. It's shared global state that (though synchronized) may still interact in unpredictable ways. It also makes test names not reproducible (though since you're throwing the system time in there, that's out the window anyways).

If you didn't eliminate this, you should at least make the def private, and possibly put it into another object that encapsulates this. This seems like a purely internal function, not meant to be user-invoked or accessible.